### PR TITLE
replace button by loading button from MUI

### PR DESCRIPTION
**Pull request by Med1233**
## Description

This one is for replacing the save button with the new Loading button presented by Material UI, which has a built-in loader 

https://user-images.githubusercontent.com/109956462/195056941-27985b7f-79f4-41a5-b51f-614a4dbb2985.mp4

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x (master)

<h2> How this pull request can be tested ? </h2>

Create a BA and click on “Save”
Check if button display loading icon before to be disabled

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: packages/centreon-ui/src/Button/Save/Content.tsx [hunk 1]
```diff
@@ -1,47 +1,30 @@
...
-  labelLoading,
-  loading,
-  iconSize,
   labelSave,
+  loading,
+  labelLoading,
 }: Props): JSX.Element | string => {
   const { t } = useTranslation();
 
+  if (loading) return t(labelLoading);
```
Comment (Thebarda) [inline packages/centreon-ui/src/Button/Save/Content.tsx:23]:
```suggestion
  if (loading) {
    return t(labelLoading);
  }
```

**Review by Thebarda — COMMENTED** [review]
You can replace the style implementation by this `className={cx({ [classes.loadingButton]: !hasLabel })}` and the `loadingButton` is made of `{
    width: theme.spacing(5),
  },`.

In the `StartIcon.tsx` file, replace the CircularProgress line by this `[propEq('loading', true), always(<SaveIcon />)],` thus you can remove all properties except `startIconConfig`

File: packages/centreon-ui/src/Button/Save/Content.tsx [hunk 2]
```diff
@@ -1,47 +1,32 @@
...
-  iconSize,
   labelSave,
+  loading,
+  labelLoading,
 }: Props): JSX.Element | string => {
   const { t } = useTranslation();
 
+  if (loading) {
+    return t(labelLoading);
+  }
```
Comment (Thebarda) [inline packages/centreon-ui/src/Button/Save/Content.tsx:25]:
```suggestion
  }
  
```

File: packages/centreon-ui/src/Button/Save/index.stories.tsx [hunk 3]
```diff
@@ -54,6 +54,10 @@ export const loadingWithTextAndSmallSize = (): JSX.Element => (
   <ButtonSave loading labelLoading="Loading" size="small" />
 );
 
+export const loadingWithTextAndSLargeSize = (): JSX.Element => (
```
Comment (Thebarda) [inline packages/centreon-ui/src/Button/Save/index.stories.tsx:57]:
```suggestion
export const loadingWithTextAndLargeSize = (): JSX.Element => (
```

**Review by Thebarda — APPROVED** [review]

**Review by JKancel — APPROVED** [review]

**Review by JKancel — APPROVED** [review]

**Pull request merged**